octobot_trading/personal_data/portfolios/portfolio_value_holder.py

```diff
@@ -60,25 +61,28 @@ def update_origin_crypto_currencies_values(self, symbol, mark_price):
         """
         currency, market = symbol_util.parse_symbol(symbol).base_and_quote()
         # update origin values if this price has relevant data regarding the origin portfolio (using both quote and base)
+        origin_crypto_currencies_with_values \
+            = set(self.origin_crypto_currencies_values.keys())
         origin_currencies_should_be_updated = (
-                (
-                        currency not in set(self.origin_crypto_currencies_values.keys()) and
-                        market == self.portfolio_manager.reference_market
-                )
-                or
-                (
-                        market not in set(self.origin_crypto_currencies_values.keys()) and
-                        currency == self.portfolio_manager.reference_market
-                )
+                currency not in  origin_crypto_currencies_with_values
+                or market not in origin_crypto_currencies_with_values
         )
+        self.last_prices_by_trading_pair[symbol] = mark_price
         if origin_currencies_should_be_updated:
-            # will fail if symbol doesn't have a price in self.origin_crypto_currencies_values and therefore
-            # requires the origin portfolio value to be recomputed using this price info in case this price is relevant
+            # will fail if symbol doesn't have a price in 
+            #   self.origin_crypto_currencies_values and therefore
+            # requires the origin portfolio value to be recomputed 
+            #   using this price info in case this price is relevant
             if market == self.portfolio_manager.reference_market:
                 self.origin_crypto_currencies_values[currency] = mark_price
+            elif currency == self.portfolio_manager.reference_market:
+                self.origin_crypto_currencies_values[market] = (
+                    constants.ONE / mark_price)
             else:
-                self.origin_crypto_currencies_values[market] = constants.ONE / mark_price
-        self.last_prices_by_trading_pair[symbol] = mark_price
+                converted_value = self._try_converting_with_multiple_pairs(
+                    currency, constants.ONE)
+                if converted_value:
```

Comment:
What should we do when `if converted_value` is False?

Reply:
It will not set a origin_crypto_currencies_value and will raise errors as it did before.

Reply:
I think we should check `if converted_value is None` to allow 0 as a value